claimManagement/src/main/java/org/openimis/imisclaims/network/request/BaseGraphQLRequest.java

```diff
@@ -0,0 +1,56 @@
+package org.openimis.imisclaims.network.request;
+
+import androidx.annotation.NonNull;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloClient;
+import com.apollographql.apollo.api.Operation;
+import com.apollographql.apollo.api.Query;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.exception.ApolloException;
+
+import org.openimis.imisclaims.BuildConfig;
+import org.openimis.imisclaims.network.apollo.DateCustomTypeAdapter;
+import org.openimis.imisclaims.network.apollo.DateTimeCustomTypeAdapter;
+import org.openimis.imisclaims.network.apollo.DecimalCustomTypeAdapter;
+import org.openimis.imisclaims.network.util.OkHttpUtils;
+import org.openimis.imisclaims.type.CustomType;
+
+public abstract class BaseGraphQLRequest {
+
+    private static final String URI = BuildConfig.API_BASE_URL+"api/graphql";
+
+    private static final ApolloClient apolloClient = ApolloClient.builder()
+            .okHttpClient(OkHttpUtils.getDefaultOkHttpClient())
+            .serverUrl(URI)
+            .addCustomTypeAdapter(CustomType.DATE, new DateCustomTypeAdapter())
+            .addCustomTypeAdapter(CustomType.DATETIME, new DateTimeCustomTypeAdapter())
+            .addCustomTypeAdapter(CustomType.DECIMAL, new DecimalCustomTypeAdapter())
+            .build();
+
+    @NonNull
+    protected <T extends Operation.Data> Response<T> makeSynchronous(Query<T,?,?> query) throws Exception {
+        final Exception[] exceptions = new Exception[1];
+        final Response<T>[] responses = new Response[1];
+        apolloClient.query(query).enqueue(new ApolloCall.Callback() {
+            @Override
+            public void onResponse(@NonNull Response response) {
+                responses[0] = response;
+            }
+
+            @Override
+            public void onFailure(@NonNull ApolloException e) {
+                exceptions[0] = e;
+            }
+        });
+
+        while (exceptions[0] == null && responses[0] == null) {
+            Thread.sleep(100);
+        }
```

Comment:
If possible I'd find another way than using a `Thread.sleep` (how does it affect the Android app? is it already in a Bg service?) especially when looping on captured references (it feels like a good recipe for a deadlock). Is there a reason you don't want to go for async future execution?

Reply:
This must be run in a `Thread`; it seems I forgot to add the  `@WorkerThread` annotation to indicate it (though the name is clear). I had to do that because Apollo doesn't expose any synchronous code. I could have used a Mutex, but it could have lead to a dead lock (if the asynchronous code finishes before the `wait` is called. I found this the cleaner solution to make the code synchronous.

I wanted the code synchronous to change as little code as possible in the original code base.

Reply:
I like the new version :+1: